lib/caesar.rb

```diff
@@ -1,6 +1,79 @@
 class Caesar
-  attr_reader :key
+  def rotation_key(key)
+    rotation = ('a'..'z').to_a
+    static = ('a'..'z').to_a
+    shifted = rotation.rotate(-key)
+    static.zip(shifted)
+  end
+
+  def unrotate_key(key)
+    rotation = ('a'..'z').to_a
+    static = ('a'..'z').to_a
+    shifted = rotation.rotate(key)
+    static.zip(shifted)
+  end
+
+  def eng_to_cipher(message, key = 3)
+    capitals = ('A'..'Z').to_a
+    encrypted = message.chars.map do |letter|
+      if letter == ' '
+        letter
+      elsif capitals.include?(letter)
+        rotate_capital(letter, key)
+      else
+        rotate(letter, key)
+      end
+    end
+    encrypted.join('')
+  end
+
+  def rotate_capital(letter, key)
+    lower = letter.downcase
+    index = 0
+    index += 1 until lower == rotation_key(key)[index][0]
+    rotation_key(key)[index][1].upcase
+  end
+
+  def rotate(letter, key)
+    index = 0
+    index += 1 until letter == rotation_key(key)[index][0]
+    rotation_key(key)[index][1]
+  end
+
+  def cipher_to_eng(message, key = 3)
+    capitals = ('A'..'Z').to_a
+    encrypted = message.chars.map do |letter|
+      if letter == ' '
+        letter
+      elsif capitals.include?(letter)
+        unrotate_capital(letter, key)
+      else
+        unrotate(letter, key)
+      end
+    end
+    encrypted.join('')
+  end
+
+  def unrotate_capital(letter, key)
```

Comment:
You are violating the DRY principle because this is very similar to unrotate. You are already making a distinction between capitals and lowercases in your map on line 48. Think about how you could use a general unrotate method to work for both capitals and lowercases.